src/worker/tasks/chainIndexer.ts

```diff
@@ -279,73 +280,75 @@ export const createChainIndexerTask = async (
           let lastIndexedBlock;
           try {
             lastIndexedBlock = await getBlockForIndexing({ chainId, pgtx });
+
+            // RPC may provide incomplete data for new blocks from the previous scan.
+            // Re-check the last few blocks to capture any missing onchain data.
+            if (lastIndexedBlock > 0) {
+              lastIndexedBlock -= NUM_BLOCKS_TO_RECHECK;
+            }
```

Comment:
Potential fix 4: Recheck the last 2 blocks to capture incomplete data from the last check.